### ISSUE TITLE
FTransUNet：复现问题

### ISSUE THREAD

**Issue by mushroom-zzx** [sstary/SSRS#133]
您好，我在运行时出现了这个警告：
D:\Miniconda\envs\FTransUNet\lib\site-packages\torch\optim\lr_scheduler.py:192: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn(
请问这个警告是需要将下面这两行移动到for e in range(1, epochs + 1)这个循环的末尾吗：     
 if scheduler is not None:
    scheduler.step()

<img width="1621" height="1857" alt="Image" src="https://github.com/user-attachments/assets/c3edee0b-ebb9-48be-bd5e-290bebaafcff" />

**Comment by mushroom-zzx** [comment]
作者您好，我读了您的FTransUNet和MFNet文章，还有几个地方存在疑惑，想向您请教一下，非常感谢：
关于MFNet：
1. MFNet本质解决的是“使SAM能够胜任多模态遥感语义分割这一新任务”这个问题吗，而不是解决多模态遥感的泛化问题，这篇文章和泛化有没有关系呢
2. MFNet的实验结果，比如下面的这张表，训练和测试都是使用的同一个数据集吗，比如将VAIHINGEN数据集划分成训练集和测试集两部分，在训练的时候对模型进行微调，然后用测试集进行测试，论文中表里的实验结果是测试集的结果，是这样吗

<img width="1583" height="700" alt="Image" src="https://github.com/user-attachments/assets/ea16f278-2157-47c9-9686-b7c5a9cfa039" />

关于数据集的问题：
3. FTransUNet中用于训练的VAIHINGEN数据集，输入模型中的dsm数据是不是归一化之前的，我用matlab打印数值，发现每个值都在245-290的区间，这个数字的单位是米吗，是表示地形＋地物的高度吗。归不归一化对实验结果有影响吗

<img width="1208" height="762" alt="Image" src="https://github.com/user-attachments/assets/98cce8ec-ba43-49a6-9049-648e82f03bfa" />

4. DSM与光学图像是不是完全对齐的，就是说光学图像中的每一个像素点是不是都有对应的DSM数值
5. 如果光学图像与DSM图像是完全对齐的话，那为什么不直接在通道上进行相连，为啥必须设计融合的模块，这里还是没太懂

**Comment by sstary** [comment]
你好，

  1.
我们解决的是将SAM应用到遥感的多模态任务中，并未探究泛化性的问题，但是实验结果也证明了SAM具有良好的泛化性，因为我们的模型只是增加了微调机制。泛化性的研究有其他的研究，你可以搜索"VFM, Cross domain"等关键词进行检索。
  2.
是的，我们的实验设置相对简单，只有训练集和测试集，没有验证集，CV，RS领域大量的工作也都采用了这种简单的机制。
  3.
DSM数据在输入到模型之前进行了归一化，实际数值和完整大图有关，具体你可以查询数据集网站：https://www.isprs.org/resources/datasets/benchmarks/UrbanSemLab/2d-sem-label-vaihingen.aspx. 不归一化肯定会有影响，但是我并没有试过。
  4.
数据集信息同样可以参照问题3的网站，该数据集基本视为完全配准之后的标准多模态数据集。
  5.
当前融合的方法有 直接相加，通道拼接，空间、通道注意力，交叉注意力机制等。简单来讲，不同模态的信息是具有互补性及干扰性的，直接相加，或者拼接，模型都难以去判断“最”有效的信息，因此，我们才需要设计复杂的融合机制。

祝好，
献平
________________________________

**Comment by mushroom-zzx** [comment]
感谢您的回复，还有几个小问题：
1. DSM数据在输入到模型之前进行了归一化，归一化是不是把DSM的数值映射到0-1的范围
2. ”不同模态的信息是具有互补性及干扰性的”这里的干扰是指什么和什么有干扰啊，互补指互补什么
3. DSM除了可以补充高度信息，还能补充别的信息吗
4. MFNet的训练和测试大概需要多长时间
